### PR TITLE
Use Exporters directly

### PR DESCRIPTION
**Pull request by hossainemruz**
Fixes: https://github.com/kubedb/project/issues/252
Fixes: https://github.com/kubedb/project/issues/250

### CODE REVIEW

File: pkg/controller/statefulset.go [hunk 1]
```diff
@@ -21,6 +22,7 @@ import (
 
 const (
 	CONFIG_MOUNT_PATH = "/elasticsearch/custom-config"
```
Comment (tamalsaha) [inline pkg/controller/statefulset.go]: Nit: Proper go style

`ConfigMountPath`

File: pkg/controller/statefulset.go [hunk 2]
```diff
@@ -487,43 +493,66 @@ func (c *Controller) upsertMonitoringContainer(statefulSet *apps.StatefulSet, el
...
-			Name: "secret",
-			VolumeSource: core.VolumeSource{
-				Secret: &core.SecretVolumeSource{
-					SecretName: elasticsearch.Spec.DatabaseSecret.SecretName,
+			volumeMounts := container.VolumeMounts
+			volumeMounts = core_util.UpsertVolumeMount(volumeMounts, certVolumeMount)
+			container.VolumeMounts = volumeMounts
+
+			volume := core.Volume{
+				Name: "exporter-certs",
```
Comment (tamalsaha) [inline pkg/controller/statefulset.go]: Can't change volume names

File: pkg/controller/statefulset.go (hunk 1 above)
Comment (hossainemruz) [reply pkg/controller/statefulset.go]: refactored

File: pkg/controller/statefulset.go [hunk 3]
```diff
@@ -487,43 +493,73 @@ func (c *Controller) upsertMonitoringContainer(statefulSet *apps.StatefulSet, el
...
-			if container.Name == "exporter" {
-				statefulSet.Spec.Template.Spec.Containers[i].Env = core_util.UpsertEnvVars(container.Env, envList...)
-				return statefulSet
+		if elasticsearch.Spec.EnableSSL {
+			certVolumeMount := core.VolumeMount{
+				Name:      "exporter-certs",
+				MountPath: ExporterCertDir,
 			}
+
+			volumeMounts := container.VolumeMounts
```
Comment (tamalsaha) [inline pkg/controller/statefulset.go]: Use online. I find that more readable.

```
core_util.UpsertVolumeMount(container.VolumeMounts, certVolumeMount)
```

If it gets too long, you can wrap it like this

```
core_util.UpsertVolumeMount(
	container.VolumeMounts,
	certVolumeMount)
```

File: pkg/controller/statefulset.go [hunk 4]
```diff
@@ -487,43 +493,73 @@ func (c *Controller) upsertMonitoringContainer(statefulSet *apps.StatefulSet, el
...
+			esCaFlag := "--es.ca=" + filepath.Join(ExporterCertDir, "root.pem")
+			args := container.Args
+
+			if len(args) == 0 || args[len(args)-1] != esCaFlag {
+				args = append(args, esCaFlag)
+			}
+			container.Args = args
 		}
+
+		containers := statefulSet.Spec.Template.Spec.Containers
```
Comment (tamalsaha) [inline pkg/controller/statefulset.go]: Same here

**Pull request merged**